rusk/src/lib/services/network.rs

```diff
@@ -0,0 +1,132 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Public Key infrastructure service implementation for the Rusk server.
+
+use kadcast::{MessageInfo, NetworkListen, Peer};
+use tokio::sync::broadcast::{self, error::RecvError, Sender};
+use tonic::{Request, Response, Status};
+use tracing::{error, info, warn};
+
+pub use super::rusk_proto::{
+    network_server::{Network, NetworkServer},
+    BroadcastMessage, Message, MessageMetadata, Null, SendMessage,
+};
+use futures::Stream;
+use std::{net::SocketAddr, pin::Pin};
+
+pub struct RuskNetwork {
+    peer: Peer,
+    sender: Sender<(Vec<u8>, SocketAddr, u8)>,
+}
+
+impl RuskNetwork {
+    pub fn new(
+        public_addr: String,
+        listen_addr: Option<String>,
+        bootstrap: Vec<String>,
+    ) -> RuskNetwork {
+        let grpc_sender = broadcast::channel(100).0;
```

Comment:
We create a channel, but discard the receiver. Can we put a comment here on why that isn't a problem?